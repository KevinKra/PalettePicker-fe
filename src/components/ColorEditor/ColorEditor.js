import React, { Component } from "react";
import "./ColorEditor.scss";

export default class ColorEditor extends Component {
  state = {
    color: "",
    updated: false
  };

  componentDidMount() {
    this.setState({ color: this.props.palette.hex });
  }

  handleChange = e => {
    const { value } = e.target;
    this.setState({ color: value, updated: true });
  };

  handleUpdate = () => {
    //handleColor passed from SavePopup container
    this.props.handleColor(this.props.palette, "update", this.state.color);
    this.setState({ updated: false });
  };

  render() {
    return (
      <article className="ColorEditor" key={this.props.palette.hex}>
        <section className="palette-inputs">
          <div
            className="palette-color"
            style={{
              backgroundColor: this.state.color
            }}
          ></div>
          <input
            type="text"
            // pattern="{7}"
            // required
            // title="7 characters"
            placeholder="#XXXXXX"
            className="color-input"
            value={this.state.color}
            onChange={e => this.handleChange(e)}
          />
        </section>
        <section className="palette-btns">
          <button
            className="update-btn"
            type="button"
            onClick={this.handleUpdate}
          >
            Update Color
          </button>
          <button
            className="remove-btn"
            type="button"
            onClick={() => this.props.handleColor(this.props.palette, "remove")}
          >
            <i className="fas fa-lg fa-trash"></i>
          </button>
        </section>
      </article>
    );
  }
}
