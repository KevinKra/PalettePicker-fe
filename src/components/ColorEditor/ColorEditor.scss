@import "../../styles/main.scss";

.ColorEditor {
	display: flex;
	justify-content: space-between;
	margin: 0.2rem 0.7rem 0 0.7rem;
	height: 2.4rem;
	background-color: $color-primary;
	border: 1px solid $color-primary;

	.palette-inputs {
		display: flex;
		width: 40%;
	}

	.palette-color {
		width: 60%;
	}

	.color-input {
		width: 40%;
		border: none;
	}

	.palette-btns {
		display: flex;
		width: 60%;
		align-items: center;
		flex: 1;
	}

	.remove-btn {
		flex: 1;
		background-color: $color-primary;
		border: none;
		width: 30%;
		margin: 0;
	}

	.remove-btn > i {
		color: $color-primary;
	}

	.update-btn {
		flex: 3;
		border-radius: 0;
		border: none;
		margin: 0;
		width: 70%;
		height: 100%;
		font-size: 10px;
		&:hover {
			background-color: red;
		}
	}
}
