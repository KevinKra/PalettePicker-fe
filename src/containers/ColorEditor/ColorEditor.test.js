import React from "react";
import ColorEditor from "./ColorEditor";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

describe("ColorEditor", () => {
  let wrapper;
  let ColorEditorInstance;
  const mockPalette = { hex: "#123" };
  const handleColorMock = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <ColorEditor palette={mockPalette} handleColor={handleColorMock} />
    );
    ColorEditorInstance = wrapper.instance();
  });
  it("should match the snapshot", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it("componentDidMount: should pass the prop hex value to color in state", () => {
    expect(ColorEditorInstance.state.color).toBe("#123");
  });
  it("handleChange: should update state.color to the value argument passed in", () => {
    ColorEditorInstance.handleChange({ target: { value: "a" } });
    expect(ColorEditorInstance.state.color).toEqual("a");
  });
  it("should invoke handleChange on 'color-input' change event", () => {
    const input = wrapper.find(".color-input");
    jest.spyOn(wrapper.instance(), "handleChange");
    input.simulate("change", { target: { value: "hello" } });
    expect(wrapper.instance().handleChange).toHaveBeenCalled();
  });
  it("should invoke the handleColor function when the 'update-btn' button is clicked", () => {
    const button = wrapper.find(".update-btn");
    button.simulate("click");
    expect(handleColorMock).toHaveBeenCalled();
  });
  it("should invoke the handleColor function when the 'remove-btn' button is clicked", () => {
    const button = wrapper.find(".remove-btn");
    button.simulate("click");
    expect(handleColorMock).toHaveBeenCalled();
  });
});
