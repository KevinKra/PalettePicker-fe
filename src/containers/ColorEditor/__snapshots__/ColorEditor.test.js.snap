// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorEditor should match the snapshot 1`] = `
<article
  className="ColorEditor"
  key="#123"
>
  <section
    className="palette-inputs"
  >
    <div
      className="palette-color"
      style={
        Object {
          "backgroundColor": "#123",
        }
      }
    />
    <input
      className="color-input"
      onChange={[Function]}
      placeholder="#XXXXXX"
      type="text"
      value="#123"
    />
  </section>
  <section
    className="palette-btns"
  >
    <button
      className="update-btn"
      onClick={[Function]}
      type="button"
    >
      Update Color
    </button>
    <button
      className="remove-btn"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fas fa-lg fa-trash"
      />
    </button>
  </section>
</article>
`;
